types/node/fs/promises.d.ts

```diff
@@ -951,7 +955,7 @@ declare module 'fs/promises' {
                   encoding: 'buffer';
               })
             | 'buffer'
-    ): AsyncIterable<Buffer>;
+    ): AsyncIterable<FileChangeInfo>;
```

Comment:
I believe we know this will be buffer, please make `FileChangeInfo` generic accordingly.

Reply:
Thank you!

Why it is a buffer?

According to the document description, fsPromises.watch returns an object containing eventType field and filename field.

When I use this api, it always returns the object just mentioned.

What documents did I miss?

How can I modify it?

Reply:
This overload is specifically created to guarantee that the `filename` will be buffer.

You should change your `FileChangeInfo` to `FileChangeInfo<T = strings | Buffer>` and use `FileChangeInfo<Buffer>` here.

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55033/commits/f11f769e07559594fa164dace78944a1c60c3c95

Thank you very much for your advice.

Is everything all right now?